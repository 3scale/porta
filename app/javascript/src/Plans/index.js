// @flow

export * from './components/DefaultPlanSelect'
export * from './components/DefaultPlanSelectCard'
export * from './components/DefaultPlanSelectWrapper'
// $FlowIgnore[cannot-resolve-module] name-mapper is replacing 'Plans'
export * from './components/ApplicationPlansTable'
// $FlowIgnore[cannot-resolve-module] name-mapper is replacing 'Plans'
export * from './components/ApplicationPlansTableCard'
// $FlowIgnore[cannot-resolve-module] name-mapper is replacing 'Plans'
export * from './components/ApplicationPlansTableCardWrapper'
