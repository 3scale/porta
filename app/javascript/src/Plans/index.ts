export { ChangePlanSelectCard } from './components/ChangePlanSelectCard'
export { DefaultPlanSelectCard } from './components/DefaultPlanSelectCard'
