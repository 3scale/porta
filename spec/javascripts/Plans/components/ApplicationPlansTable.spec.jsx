// @flow

import React from 'react'
import { mount } from 'enzyme'

import { ApplicationPlansTable } from 'Plans'

const plans = []
const defaultProps = {
  onAction: () => {},
  plans,
  count: plans.length,
  searchHref: '/plans'
}

const mountWrapper = (props) => mount(<ApplicationPlansTable {...{ ...defaultProps, ...props }} />)

afterEach(() => {
  jest.resetAllMocks()
})

it('should render itself', () => {
  const wrapper = mountWrapper()
  expect(wrapper.exists()).toBe(true)
})

it('should render a table with Name, Applications and State', () => {
  const wrapper = mountWrapper()
  expect(wrapper.find('th')).toMatchSnapshot()
})
