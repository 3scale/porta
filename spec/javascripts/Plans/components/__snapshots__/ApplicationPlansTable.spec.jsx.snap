// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a table with Name, Applications and State 1`] = `
Array [
  <th
    className=""
    data-key={0}
    data-label="Name"
    onMouseEnter={[Function]}
    scope="col"
  >
    Name
  </th>,
  <th
    className=""
    data-key={1}
    data-label="Applications"
    onMouseEnter={[Function]}
    scope="col"
  >
    Applications
  </th>,
  <th
    className=""
    data-key={2}
    data-label="State"
    onMouseEnter={[Function]}
    scope="col"
  >
    State
  </th>,
]
`;
